specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/playbookConfigurations.json

```diff
@@ -0,0 +1,461 @@
+        },
+        "createdBy": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The AAD object id that created the playbook configuration."
+        },
+        "lastUpdatedDateTimeUtc": {
+          "readOnly": true,
+          "type": "string",
+          "format": "date-time",
+          "description": "The playbook configuration last updated date."
+        },
+        "lastUpdatedBy": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The AAD object id that last updated the playbook configuration."
+        }
+      }
+    },
+    "PlaybookConfigurationSource": {
+      "type": "object",
+      "description": "The source event types which evaluate the playbook configuration set of rules. For example - security alerts and recommended tasks.",
+      "properties": {
+        "eventSource": {
+          "type": "string",
+          "description": "A valid event source type."
+        },
+        "ruleSets": {
+          "type": "array",
+          "description": "A set of rules which evaluate upon event interception.",
+          "items": {
+            "$ref": "#/definitions/PlaybookConfigurationRuleSet"
+          }
+        }
+      }
+    },
+    "PlaybookConfigurationAction": {
+      "type": "string",
+      "description": "The uri that should be triggered.",
+      "format": "uri"
+    },
+    "PlaybookConfigurationRuleSet": {
+      "type": "object",
+      "description": "A rule set which evaluates all its rules upon an event interception.",
+      "properties": {
+        "rules": {
+          "type": "array",
+          "$ref": "#/definitions/PlaybookConfigurationTriggeringRule"
+        }
+      }
+    },
+    "PlaybookConfigurationTriggeringRule": {
+      "type": "object",
+      "description": "A logic rule which evaluates upon event interception. The rule is configured by comparing the specified expected value, in an expected property within the event model, by the specified operator.",
+      "properties": {
+        "propertyJPath": {
+          "type": "string",
+          "description": "The JPath of the entity model property that should be checked."
+        },
+        "propertyType": {
+          "type": "string",
+          "description": "The data type of the compared operands."
+        },
+        "expectedValue": {
+          "type": "string",
+          "description": "The expected value."
+        },
+        "conditions": {
+          "type": "string",
+          "description": "A valid comparer operator to use."          
+        }
+      }
+    },
+    "PlaybookConfigurationValidationStatus": {
+      "type": "object",
+      "description": "The playbook configuration model state property bag.",
+      "properties": {
+        "isValid": {
+          "type": "boolean",
+          "description": "Indicates whether the model is valid or not."
+        },
+        "message": {
```

Comment:
maybe you can add a list of all the failed validations?
this way the user will reduce the number of request count before a success

Reply:
This validations model will be designed for the PP version. At the meantime, for pp, all validation errors will be concatenated to a single string. Is it ok?

Reply:
It sounds like you're planning to return 200 OK with isValid set to false for invalid playbooks. You should return 200 OK only if the validation is successful. For validation failures, you should return 400 Bad Request and the error information should be captured inside the standard ARM error response. Take a look at "Error Response Content" section at [https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md](https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md). 

---
In reply to: [273947475](https://github.com/Azure/azure-rest-api-specs/pull/5548#discussion_r273947475) [](ancestors = 273947475)